go-controller/pkg/util/util.go

```diff
@@ -317,6 +318,37 @@ func IsClusterIP(svcVIP string) bool {
 	return false
 }
 
+// GetActiveNetworkForNamespace returns the active network for the given namespace
+// active network here means the network managing this namespace and responsible for
+// plumbing all the entities for this namespace
+// this is:
+// 1) "default" if there are no NADs in the namespace OR all NADs are primaryNetwork:false
+// 2) "<secondary-network-name>" if there is exactly ONE NAD with primaryNetwork:true
+// 3) "unknown" under all other conditions
+func GetActiveNetworkForNamespace(nads []*nadapi.NetworkAttachmentDefinition, podNamespace string) (string, error) {
+	var activeNetwork string
+	if len(nads) == 0 {
+		return types.DefaultNetworkName, nil
+	}
+	numberOfPrimaryNetworks := 0
+	for _, nad := range nads {
+		nadInfo, err := ParseNADInfo(nad)
+		if err != nil {
+			return activeNetwork, err
+		}
+		if nadInfo.IsPrimaryNetwork() {
+			activeNetwork = nadInfo.GetNetworkName()
+			numberOfPrimaryNetworks++
```

Comment:
ack I'll split out the counting as a separate thing and do your suggestion ^

Reply:
ah no parsed NAD info is what contains that information and the helper util GetNetworkName() is available only for the parsed NAD which has to happen inside the loop :(

Reply:
Then we store the parsed nads instead of a counter
```golang
primaryNetworks := []*NetInfo{}
	for _, nad := range namespaceNADs {
		nadInfo, err := ParseNADInfo(nad)
		if err != nil {
			return activeNetwork, err
		}
		if nadInfo.IsPrimaryNetwork() {
			primryNetworks = append(primaryNetworks, nadInfo)
...
if len(primaryNetworks) > 1 

...
primaryNetworks[0].GetNetworkName()
...